src/Xamarin.Android.Build.Tasks/Utilities/ResourceIdentifier.cs

```diff
@@ -16,18 +16,18 @@ public class ResourceIdentifier {
 		private const string FormattingCharacter = @"\p{Cf}";
 		private const string ConnectingCharacter = @"\p{Pc}";
 		private const string DecimalDigitCharacter = @"\p{Nd}";
-		private const string CombiningCharacter = @"\p{Mn}\p{Mc}";
-		private const string LetterCharacter = @"\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Lo}\p{Nl}";
+		private const string CombiningCharacter = @"\p{Mn},\p{Mc}";
+		private const string LetterCharacter = @"\p{Lu},\p{Ll},\p{Lt},\p{Lm},\p{Lo},\p{Nl}";
 
-		private const string IdentifierPartCharacter = LetterCharacter +
-			DecimalDigitCharacter +
-			ConnectingCharacter +
-			CombiningCharacter +
+		private const string IdentifierPartCharacter = LetterCharacter + "," +
```

Comment:
I believe we *don't* want to add `","` to this list or the previous lists:

```csharp
string FormattingCharacter = @"\p{Cf}";
string ConnectingCharacter = @"\p{Pc}";
string DecimalDigitCharacter = @"\p{Nd}";
string CombiningCharacter = @"\p{Mn},\p{Mc}";
string LetterCharacter = @"\p{Lu},\p{Ll},\p{Lt},\p{Lm},\p{Lo},\p{Nl}";
string IdentifierPartCharacter = LetterCharacter + "," +
	DecimalDigitCharacter + "," + ConnectingCharacter + "," +
	CombiningCharacter + "," + FormattingCharacter;
string IdentifierStartCharacter = "(" + LetterCharacter + ",_)";
string Identifier = IdentifierStartCharacter + "((" + IdentifierPartCharacter + ")+)*";
var r = new Regex ($"[^{Identifier}]", RegexOptions.Compiled);
r.Replace ("hello,world", "_");
// "hello,world"
```

Note that the `,` isn't replaced.

If we fixup *all* these declarations so that `,` is *not* included, then it replaces as I'd expect:

```csharp
string FormattingCharacter = @"\p{Cf}";
string ConnectingCharacter = @"\p{Pc}";
string DecimalDigitCharacter = @"\p{Nd}";
string CombiningCharacter = @"\p{Mn}\p{Mc}";
string LetterCharacter = @"\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Lo}\p{Nl}";
string IdentifierPartCharacter = LetterCharacter +
	DecimalDigitCharacter + ConnectingCharacter +
	CombiningCharacter + FormattingCharacter;
string IdentifierStartCharacter = "(" + LetterCharacter + "_)";
string Identifier = IdentifierStartCharacter + "((" + IdentifierPartCharacter + ")+)*";
var r = new Regex ($"[^{Identifier}]", RegexOptions.Compiled);
r.Replace ("hello_world", "_");
```

Reply:
Ah crap! the rebase broke stuff :( I'll fix it